README.md

```diff
@@ -58,6 +60,15 @@ We use --recursive flag for our dependency on googletest framework.
 
 First create a build directory from the root of the project. In the build directory, run the following commands depending on your environment.
 
+For compiling with SSL support: 
+- You need have the openssl installed in your development environment. Furthermore, the openssl include directory should be added to include directories and the openssl library directory should be listed in the link directories. Furthermore you need to set the openssl libraries to link.
+ 
+- You can provide your openssl installation directory to cmake using the following flags: -DHZ_OPENSSL_INCLUDE_DIR="Path to open installation include directory" and -DHZ_OPENSSL_LIB_DIR="Path to openssl lib directory"
+
+- Use -DHZ_COMPILE_WITH_SSL=ON
+
+- Check top level CMakeLists.txt file to see which libraries we link for openssl in which environment. For Mac OS and Linux, we link with "ssl" and "crypto", for windows and example linkage for 64 bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".
+
```

Comment:
Check the top level `CMakeLists.txt` file to see which libraries we link for openssl in which environment. For Mac OS and Linux, we link with "ssl" and "crypto", for Windows there is an example linkage for 64 bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".